gtda/utils/validation.py

```diff
@@ -174,3 +176,117 @@ def validate_metric_params(metric, metric_params):
+        Each entry of X corresponds to an image.
+
+    kwargs: dict or None, optional, default: ``None``
+        Parameters accepted by
+        :func:`~gtda.utils.validation.check_list_of_arrays`.
+
+    Returns
+    -------
+    X : list of ndarray, as modified by check_array
+
+    """
+    kwargs_default = {'force_same_n_axis':  False,
+                      'force_same_dim': False, 'force_all_finite': True,
+                      'ensure_2d': False, 'allow_nd': True}
+    kwargs_default.update(kwargs)
+    return check_list_of_arrays(X, **kwargs_default)
+
+
+def check_list_of_point_clouds(X, **kwargs):
+    """"Check a list of arrays representing point clouds, by integrating
+    through the input one by one. To pass a test with
+    :param:`kwargs`=``None``, all point clouds x, y in X must satisfy:
+        - x.ndim == 2,
+        - len(y.shape) == len(y.shape).
+
+    Parameters
+    ----------
+    X : list(ndarray), such that `X[i].ndim==2` (n_points, n_dimensions),
+        or an array `X.dim==3`
+
+    kwargs: dict or None, optional, default: ``None``
+        Parameters accepted by
+        :func:`~gtda.utils.validation.check_list_of_arrays`.
+
+    Returns
+    -------
+    X : list of input arrays, as modified by check_array
+
+    """
+    kwargs_default = {'ensure_2d': True, 'force_all_finite': False,
+                      'force_same_dim': False, 'force_same_n_axis': True}
+    kwargs_default.update(kwargs)
+    return check_list_of_arrays(X, **kwargs_default)
+
+
+def check_list_of_arrays(X, force_same_dim=True, force_same_n_axis=True,
+                         **kwargs):
+    """"Check a list of arrays, by integrating through the input one by one.
+    The constraints are to be specified in :param:`kwargs`. On top of
+    parameters from :func:`~sklearn.utils.validation.check_array`,
+    the optional parameters are listed below.
+
+    Parameters
+    ----------
+    X : list(ndarray), such that `X[i].ndim==2` (n_points, n_dimensions),
+        or an array `X.dim==3`
+
+    force_same_dim : bool, optional, default: ``True``
+        Indicates whether the shapes of the elements of X should all
+        be the same.
+
+    force_same_n_axis : bool, optional, default: ``True``
+        Indicates whether the number of axes in the elements of X should all
+        be the same.
+
+    kwargs: dict or None, optional, default: ``None``
+        Parameters accepted by :func:`~sklearn.utils.validation.check_array`.
+
+    Returns
+    -------
+    X : list of input arrays, as modified by check_array
+
+    """
+
+    # if restrictions on the dimensions of the input are imposed
+    if force_same_dim:
+        shapes = [x.shape for x in X]
+        if not(all([shapes[0] == s for s in shapes])):
+            raise ValueError("The arrays in X do not have the same dimensions"
+                             "({}), while they should.".format(shapes))
+    # if the number of dimensions can vary
+    if force_same_n_axis:
```

Comment:
`force_same_dim`(AKA `force_same_shape`) supersedes the  calculations here so I'd suggest restructuring the code so that no unnecessary checks are made.